cr.c

```diff
@@ -84,9 +84,11 @@ static void dill_ctx_timer_handler(int sig, siginfo_t *si, void *uc) {
 #ifndef __APPLE__
     struct dill_ctx_cr *ctx = si->si_value.sival_ptr;
     int or = timer_getoverrun(ctx->timer);
-    if(or) ctx->do_poll = 1;
+    if(or) {
+        __sync_lock_test_and_set(&ctx->do_poll, 1);
+    }
 #else
-    dill_poll_count++;
+    __sync_fetch_and_add(&dill_poll_count, 1);
```

Comment:
That should be okay then: some rough timings for perf/go with default compile options on my laptop:
CLOCK_MONOTONIC: 85 ns
CLOCK_MONOTONIC_COARSE: 65 ns
POSIX timers: 53ns

Calculating N at startup for `rdtsc` will probably need to measure a block of code with clock_gettime and compare it with the cycles count produced by rdtsc.